nodejs/eks/authenticationMode.ts

```diff
@@ -34,24 +35,31 @@ export function validateAuthenticationMode(args: ClusterOptions) {
         );
     }
 
-    const configMapOnlyProperties: (keyof ClusterOptions)[] = [
-        "roleMappings",
-        "userMappings",
-        "instanceRoles",
-    ];
-    const apiOnlyProperties: (keyof ClusterOptions)[] = ["accessEntries"];
-
     if (!supportsConfigMap(args.authenticationMode)) {
-        configMapOnlyProperties.forEach((prop) => {
-            if (args[prop]) {
+        const check: (prop: string) => (propertyValue: any|undefined) => void = (prop) => (pv) => {
+            if (pv !== undefined) {
                 throw new Error(
                     `The '${prop}' property is not supported when 'authenticationMode' is set to '${args.authenticationMode}'.`,
                 );
             }
+        };
+
+        args.roleMappings = validatedInput(args.roleMappings, check("roleMappings"));
```

Comment:
Why don't we extend this to roleMappings and userMappings as well? Would be a nice improvement and it's a bit odd imho if roleMappings/userMappings don't allow empty arrays while instanceRoles  does

Reply:
Done.